doc/wallet-storage.md

```diff
@@ -0,0 +1,255 @@
+# Wallet Storage Design
+
+In current state libindy allows to plug different wallet implementations. Plugged wallet now handles both security
```

Comment:
What about what entities should be stored in wallet vs. multiple wallets vs. some storage?
Is it up to the app where to store, for example, claims?

I think there are 3 main options possible:
1) Only secrets (private, master secret) are stored in the wallet on libindy level. All other entities are assumed to be stored on app level.
2) Everything is stored in the wallet on libindy level. Here we assume that each libindy call works with **one** wallet only, so that by providing a proper wallet_handle, the app can decide what kind of wallet/implementation needs to be used. Here the wallet may mean something more general (like just a storage with optional security properties).